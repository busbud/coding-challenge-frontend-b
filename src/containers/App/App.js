import React, { Component } from 'react'

import { Loader } from 'styled-icons/feather'

import Container from '../../components/Container'
import Departure from '../../components/Departure'
import Header from '../../components/Header'
import Locale from '../../components/Locale'

import {
  StyledDeparturesLength,
  StyledLoader
} from './styles'

class App extends Component {
  componentWillMount() {
    const { defaultParams, defaultQuery, getSchedules } = this.props

    getSchedules(defaultQuery, defaultParams)
  }


  handleChangeLocale = event => {
    const { defaultQuery, defaultParams, getSchedules, changeLocale, locales } = this.props;

    const locale = locales.find(locale => locale.lang === event.target.value)

    changeLocale(locale.currency, locale.lang)
    getSchedules(defaultQuery, defaultParams)
  }


  render() {
    const { cities, defaultQuery: { outbound_date }, departures, isLoading, locales, t } = this.props

    return (
      <>
        <Locale
          changeLocale={this.handleChangeLocale}
          locales={locales} />
        <Header date={outbound_date} />
        <Container>
          {isLoading
            ? <StyledLoader><Loader size={26} /></StyledLoader>
            : (departures.length
              ?
              <>
                <StyledDeparturesLength>{t('search.result.length', { count: departures.length })}</StyledDeparturesLength>
                {departures.map(departure => (
                  <Departure
                    cities={cities}
                    departure={departure}
                    key={departure.id} />
                ))}
              </>
              : <StyledDeparturesLength>{t('search.result.none')}</StyledDeparturesLength>
            )
          }
        </Container>
      </>
    )
  }
}

export default App
