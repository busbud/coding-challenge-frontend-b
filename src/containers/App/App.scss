@import "../../styles/buttons";
@import "../../styles/colors";
@import "../../styles/layout";
@import "../../styles/responsive_utilities";

.bosheaga-app {
  padding: 0;
}
