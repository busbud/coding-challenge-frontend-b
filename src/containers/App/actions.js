import {
  UPDATE_QUERY,
  FETCH_DEPARTURES,
  FETCH_DEPARTURES_SUCCESS,
  FETCH_DEPARTURES_ERROR,
  FETCH_DEPARTURES_COMPLETE,
} from './constants';

export function updateQuery(query) {
  return {
    type: UPDATE_QUERY,
    query,
  };
}

export function fetchDepartures() {
  return {
    type: FETCH_DEPARTURES,
  };
}

export function fetchDeparturesSuccess(xDepartures) {
  return {
    type: FETCH_DEPARTURES_SUCCESS,
    xDepartures,
  };
}

export function fetchDeparturesError() {
  return {
    type: FETCH_DEPARTURES_ERROR,
  };
}

export function fetchDeparturesComplete() {
  return {
    type: FETCH_DEPARTURES_COMPLETE,
  };
}
