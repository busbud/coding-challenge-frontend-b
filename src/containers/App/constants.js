export const LOCALES = ['en', 'fr'];

export const DEFAULT_LOCALE = 'en';

export const UPDATE_QUERY = '@@/App/UPDATE_QUERY';

export const FETCH_DEPARTURES = '@@/App/FETCH_DEPARTURES';

export const FETCH_DEPARTURES_SUCCESS = '@@/App/FETCH_DEPARTURES_SUCCESS';

export const FETCH_DEPARTURES_ERROR = '@@/App/FETCH_DEPARTURES_ERROR';

export const FETCH_DEPARTURES_COMPLETE = '@@/App/FETCH_DEPARTURES_COMPLETE';
