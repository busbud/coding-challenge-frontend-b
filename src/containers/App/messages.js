import { defineMessages } from 'react-intl';

export default defineMessages({
  error_content: {
    id: 'containers.app.error.content',
    defaultMessage: 'Our apologies an unknown error has occurred.',
  },
});
