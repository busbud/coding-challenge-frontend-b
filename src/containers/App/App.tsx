import React from 'react';

import "./App.scss";
import { Container } from "react-bootstrap";

import Header from "../Header/Header";
import Presentation from "../../components/Presentation/Presentation";
import SearchSection from "../SearchSection/SearchSection";

const App: React.FC = () => {
  return ( 
      <Container fluid className="bosheaga-app p-0">
        <Header />
        <Presentation />
        <SearchSection />
      </Container>
  );
};

export default App;
