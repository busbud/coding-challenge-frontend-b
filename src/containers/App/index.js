import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { withNamespaces } from 'react-i18next'

import {
  changeLocale as changeLocaleAction,
} from '../../store/services/locales/actions'

import {
  getSchedules as getSchedulesAction,
} from '../../store/services/schedules/actions'

import * as schedulesSelectors from '../../store/services/schedules/selectors'

import App from './App'

const locales = [{
  currency: 'USD',
  lang: 'en',
}, {
  currency: 'CAD',
  lang: 'fr',
}];

const defaultQuery = {
  origin: 'dr5reg',
  destination: 'f25dvk',
  outbound_date: '2019-08-02'
}

const defaultParams = {
  adult: 1,
  child: 0,
  senior: 0
}

const mapStateToProps = state => {
  return {
    cities: schedulesSelectors.getCities(state),
    departures: schedulesSelectors.getDepartures(state),
    defaultParams,
    defaultQuery,
    isLoading: schedulesSelectors.getLoading(state),
    locales,
  }
}

const mapDispatchToProps = dispatch => {
  return bindActionCreators({
    changeLocale: changeLocaleAction,
    getSchedules: getSchedulesAction,
  }, dispatch)
}

const AppContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(App)

export default withNamespaces('translations')(AppContainer)
