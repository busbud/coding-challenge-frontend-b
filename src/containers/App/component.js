import React from 'react';
import { object, bool } from 'prop-types';
import { FormattedMessage } from 'react-intl';
import Reboot from 'material-ui/Reboot';
import Grid from 'material-ui/Grid';
import Snackbar from 'material-ui/Snackbar';
import OnBoarding from '../../pages/OnBoarding';
import Header from '../../components/Header';
import messages from './messages';

const propTypes = {
  classes: object.isRequired,
  hasError: bool.isRequired,
};
const Component = ({ classes, hasError }) => (
  <div className={classes.root}>
    <Reboot />
    <Grid container justify="center">
      <Grid item xs={12}>
        <Header />
        <OnBoarding />
        <Snackbar open={hasError} message={<FormattedMessage {...messages.error_content} />} />
      </Grid>
    </Grid>
  </div>
);

Component.propTypes = propTypes;

export default Component;
