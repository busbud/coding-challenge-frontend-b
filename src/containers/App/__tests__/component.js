import React from 'react';
import { shallow } from 'enzyme';
import Snackbar from 'material-ui/Snackbar';
import messages from '../messages';
import App from '../component';

const props = {
  classes: {
    root: 'fakeRootClass',
  },
  hasError: false,
};

describe('containers | App | component', () => {
  describe('render', () => {
    it('should render without throwing and error', () => {
      // given
      const wrapper = shallow(<App {...props} />);

      // then
      expect(wrapper).toBeDefined();
    });

    it('should match snapshot', () => {
      // given
      const wrapper = shallow(<App {...props} />);

      // then
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Root', () => {
    it('should have proper className props', () => {
      // given
      const wrapper = shallow(<App {...props} />);
      const className = wrapper.props().className;

      // then
      expect(className).toBe(props.classes.root);
    });
  });

  describe('Snackbar', () => {
    it('should have proper open props', () => {
      // given
      const customProps = {
        ...props,
        hasError: true,
      };
      const wrapper = shallow(<App {...customProps} />);
      const open = wrapper.find(Snackbar).props().open;

      // then
      expect(open).toBe(true);
    });
    it('should have proper message props', () => {
      // given
      const wrapper = shallow(<App {...props} />);
      const message = wrapper.find(Snackbar).props().message.props;

      // then
      expect(message).toMatchObject(messages.error_content);
    });
  });
});
