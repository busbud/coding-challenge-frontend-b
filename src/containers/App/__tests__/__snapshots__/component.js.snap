// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers | App | component render should match snapshot 1`] = `
<div
  className="fakeRootClass"
>
  <WithStyles(Reboot) />
  <WithStyles(GridWrapper)
    container={true}
    justify="center"
  >
    <WithStyles(GridWrapper)
      item={true}
      xs={12}
    >
      <WithStyles(Component) />
      <Component />
      <WithStyles(Snackbar)
        message={
          <FormattedMessage
            defaultMessage="Our apologies an unknown error has occurred."
            id="containers.app.error.content"
            values={Object {}}
          />
        }
        open={false}
      />
    </WithStyles(GridWrapper)>
  </WithStyles(GridWrapper)>
</div>
`;
